crates/katana/rpc/rpc/src/starknet/mod.rs

```diff
@@ -71,6 +71,7 @@ struct Inner<EF: ExecutorFactory> {
     block_producer: BlockProducer<EF>,
     blocking_task_pool: BlockingTaskPool,
     forked_client: Option<ForkedClient>,
+    max_chunk_size: u64,
```

Comment:
lets make this field into an `Option` instead. also maybe lets create a config struct, `StarknetApiConfig`, to encapsulate this value.

Reply:
we should also update all the `StarknetApi::new` functions to accept the new config struct

Reply:
make sure to update the `RpcConfig` with the `max_events_page_size` option

https://github.com/dojoengine/dojo/blob/0c70748fff67947895f3ff9255ef56ad986a5f41/crates/katana/node/src/config/rpc.rs#L20-L28

i think 'page size' is a much better name in this case. maybe we should just call it page size anywhere else and just leave the chunk size name in the server impl.